src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -889,93 +1142,147 @@ proto.clearState = function(options={}) {
   this.removeAddFeaturesLayerResultInteraction();
 };
 
+/**
+ * @FIXME add description
+ */
 proto.getState = function() {
   return this.state;
 };
 
+/**
+ * @FIXME add description
+ * 
+ * @param state 
+ */
 proto.setState = function(state) {
   this.state = state;
 };
 
+/**
+ * @FIXME add description
+ * 
+ * @param project 
+ */
 proto._setRelations = function(project) {
   const projectRelations = project.getRelations();
   this._relations = projectRelations ? _.groupBy(projectRelations,'referencedLayer'):  [];
 };
 
+/**
+ * @param layerId 
+ */
 proto.getAtlasByLayerId = function(layerId) {
   return this._atlas.filter(atlas => atlas.atlas.qgs_layer_id === layerId);
 };
 
-proto._setAtlasActions = function(project){
+/**
+ * @FIXME add description
+ * 
+ * @param project 
+ */
+proto._setAtlasActions = function(project) {
   this._atlas = project.getPrint().filter(printconfig => printconfig.atlas) || [];
 };
 
+/**
+ * @FIXME add description
+ * 
+ * @param querytitle
+ */
 proto.setTitle = function(querytitle) {
   this.state.querytitle = querytitle || "";
 };
 
+/**
+ * @FIXME add description
+ */
 proto.reset = function() {
   this.clearState();
 };
 
 /**
- * Method that convert response from Data Provider to a Query Result component data structure
+ * Converts response from DataProvider into a QueryResult component data structure
+ * 
  * @param featuresForLayers: Array contains for each layer features
+ * 
  * @returns {[]}
- * @private
  */
 proto._digestFeaturesForLayers = function(featuresForLayers) {
```

Comment:
this is a blob function, I partially rewrote it to try to make it more readable, check it carefully